src/daft-shuffles/src/client/flight_client.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use arrow_flight::{client::FlightClient, decode::FlightRecordBatchStream, Ticket};
+use common_error::{DaftError, DaftResult};
+use daft_core::{prelude::Schema, series::Series};
+use daft_recordbatch::RecordBatch;
+use daft_schema::field::Field;
+use futures::{FutureExt, Stream, StreamExt};
+use tonic::transport::Channel;
+
+pub struct ShuffleFlightClient {
+    inner: FlightClient,
+    address: String,
+}
+
+impl ShuffleFlightClient {
+    pub async fn new(address: String) -> Self {
+        let endpoint = Channel::from_shared(address.clone()).unwrap();
+        let channel = endpoint.connect().await.unwrap();
```

Comment:
usually we want to separate the `new` and the `connect` into seperate methods for better control and error handling